packages/fuels-contract/src/contract.rs

```diff
@@ -203,7 +223,45 @@ impl Contract {
             return Err(Error::InvalidData(extension.to_str().unwrap().to_owned()));
         }
         let bin = std::fs::read(binary_filepath)?;
-        Ok(CompiledContract { raw: bin, salt })
+
+        Ok(CompiledContract {
+            raw: bin,
+            salt,
+            storage_slots: Self::auto_init_storage(binary_filepath),
+        })
+    }
+
+    fn auto_init_storage(binary_filepath: &str) -> Vec<StorageSlot> {
+        let storage_path = format!(
+            "{}{}",
+            Path::new(binary_filepath)
+                .with_extension("")
+                .to_str()
+                .unwrap(),
+            "-storage_slots.json"
```

Comment:
The name of this file is not really in the spec and I'm not sure we should rely on it here. Should this maybe be a user input just like the JSON ABI file?

Reply:
I hate to ask for more inputs from the user... but I don't see a way out in this case :(

Reply:
Yeah :/ That being said, `forc new` (the new `forc init`) should and will handle that for the user, just like JSON ABI. I don't remember the last time I had to worry about pointing to a specific JSON or *.bin in an SDK test.

Reply:
Then its semi-automatic storage initialization :). I saw that when ``forc`` generates a .json file, it only adds "-storage.json" to the name of contract, so I thought why not to use that. Ok, ill change this.